### PR TITLE
fix: Updates to support OpenAI better

### PR DESCRIPTION
**Pull request by jongio**
## Description

1. OpenAI deployment names need to match what the application developer provides.  For example, if they name the deployment 'foo', then their code with use 'foo'.  The bicep shouldn't generate a name for the deployment because it won't match what the developer expects.  So, I'm changing that to deployment.name.

2. To minimize the amount of code required for the user, we'll now default the sku to Standard/20.

3. To get the 'openai.azure.com' domain you have to specify a customSubDomainName, so we'll default it to the `name` var.

4. networkAcls needs to be `null`, not `{}` if not provided.

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ x] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by shenglol** [comment]
Could you update the module version to `1.0.6` in the README examples? Sorry I missed this in the previous PRs.

File: modules/ai/cognitiveservices/main.bicep [hunk 1]
```diff
@@ -137,9 +137,12 @@ resource cognitiveService 'Microsoft.CognitiveServices/accounts@2023-05-01' = {
 
 @batchSize(1)
 resource cognitiveServiceDeployment 'Microsoft.CognitiveServices/accounts/deployments@2023-05-01' = [for (deployment, index) in deployments: {
```
Comment (dciborow) [inline modules/ai/cognitiveservices/main.bicep:139]: Can we change deployments from a generic array to an array of a a user-defined type?

**Comment by jongio** [comment]
> Could you update the module version to `1.0.6` in the README examples? Sorry I missed this in the previous PRs.

Done

File: modules/ai/cognitiveservices/main.bicep (hunk 1 above)
Comment (jongio) [reply modules/ai/cognitiveservices/main.bicep:139]: @dciborow - I created a new issue to track that change https://github.com/Azure/bicep-registry-modules/issues/516

**Review by shenglol — APPROVED** [review]

**Pull request merged**